### PR TITLE
feat: handle error if api does not exist

### LINKED ISSUE #849: lint command not working when no API is added to options
**Describe the bug**
When you have added a configuration file (`.redocly.yaml`) and defined your `apis` there, you're still unable to run `redocly lint` without other arguments. 

Example:

Working:
`redocly lint someApi@v1`
Not working:
`redocly lint`  

**To Reproduce**
1. Create `redocly.yaml` file and add a definition to the `apis` section.
2. Run this command - `redocly lint`  
4. See error:
`error: missing required argument apis.
`

**Expected behavior**
The `lint` command picks up the definitions defined in the `apis` section of the configuration file and works as expected.

**Logs**
N/a

**OpenAPI definition**
N/a

**Redocly Version(s)**
latest

**`Node.js` Version(s)**
15

**Additional context**
Some context can be found here: https://redoc-ly.slack.com/archives/C02UHE7EL3E/p1661807377826659

### PR DESCRIPTION
**Pull request by SmoliyY**
## What/Why/How?
Add error handle in cases if `definition` does not exist. Previously `cli` throw native error 
Screenshot 2022-09-26 at 17 19 41

Also there was and issue if `root` of apis provided as link. See #849 and #850

## Reference
Closes #849, #850
## Screenshots (optional)
Example with run `redocly lint resources/pets.yaml not-exist.yaml`
Screenshot 2022-09-28 at 18 21 50

## Check yourself

- [ ] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [ ] All new/updated code is covered with tests

## Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### CODE REVIEW

File: packages/cli/src/utils.ts [hunk 1]
```diff
@@ -45,6 +45,34 @@ function isNotEmptyArray<T>(args?: T[]): boolean {
   return Array.isArray(args) && !!args.length;
 }
 
+function isApiPathValid(apiPath: string): string | void {
+  if (!apiPath.trim()) {
+    exitWithError('Path can not be empty.');
```
Comment (adamaltman) [inline packages/cli/src/utils.ts:50]:
```suggestion
    exitWithError('Path cannot be empty.');
```

File: packages/cli/src/utils.ts [hunk 2]
```diff
@@ -45,6 +45,34 @@ function isNotEmptyArray<T>(args?: T[]): boolean {
...
+    return;
+  }
+  if (fs.existsSync(apiPath) || isURL(apiPath)) {
+    return apiPath;
+  } else {
+    exitWithError(
+      yellow(`\n ${apiPath} ${red(`doe's not exist or invalid. Please provide valid path.`)}`)
```
Comment (adamaltman) [inline packages/cli/src/utils.ts:57]:
```suggestion
      yellow(`\n ${apiPath} ${red(`does not exist or is invalid. Please provide a valid path.`)}`)
```

File: redocly.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+apis: 
+  main:
+    root: ./../../../do-not-exist.yaml
```
Comment (adamaltman) [inline redocly.yaml:3]:
```suggestion
    root: ./../../../does-not-exist.yaml
```

**Review by adamaltman — CHANGES_REQUESTED** [review]
Some grammar changes to the messages. This is missing tests too.

File: packages/cli/src/utils.ts [hunk 4]
```diff
@@ -323,3 +351,7 @@ export function isSubdir(parent: string, dir: string): boolean {
   const relative = path.relative(parent, dir);
   return !!relative && !/^..($|\/)/.test(relative) && !path.isAbsolute(relative);
 }
+
+export function isURL(str: string): boolean {
+  return /^(https?:)\/\//m.test(str);
```
Comment (adamaltman) [inline packages/cli/src/utils.ts:356]: What if it's an http url?

File: packages/cli/src/utils.ts [hunk 5]
```diff
@@ -25,13 +25,20 @@ export async function getFallbackApisOrExit(
...
-  if (!isNotEmptyArray(res)) {
-    process.stderr.write('error: missing required argument `apis`.\n');
+
+  const filteredInvalidEntrypoints = res.filter(({ path }) => !isApiPathValid(path));
+  if (isNotEmptyArray(filteredInvalidEntrypoints)) {
+    for (const { path } of filteredInvalidEntrypoints) {
+      process.stderr.write(
+        yellow(
+          `\n ${relative(process.cwd(), path)} ${red(
+            `doe's not exist or invalid. Please provide valid path. \n\n`
```
Comment (adamaltman) [inline packages/cli/src/utils.ts:37]:
```suggestion
            `does not exist or is invalid. Please provide a valid path. \n\n`
```

File: redocly.yaml (hunk 3 above)
Comment (SmoliyY) [reply redocly.yaml:3]: This file got here accidentally. Thx for pointing out

File: packages/cli/src/utils.ts (hunk 4 above)
Comment (SmoliyY) [reply packages/cli/src/utils.ts:350]: It will pass, but there is another check that already exist and can be reused.

**Review by adamaltman — APPROVED** [review]

**Comment by tatomyr** [comment]
Please check if the logic covers the `extends` functionality: https://redocly.com/docs/cli/configuration/#extends-list

**Review by RomanHotsiy — APPROVED** [review]

**Comment by adamaltman** [comment]
And check it covers the `plugins` paths too.

**Comment by SmoliyY** [comment]
> And check it covers the `plugins` paths too.
> Please check if the logic covers the `extends` functionality: https://redocly.com/docs/cli/configuration/#extends-list

This check do not handle both of those cases. It not hard so to add such check, but it depends if we want to proceed in cases if `plugin` or `config` does not exist. I think we should proceed with same behaviour as it is work now ( Throw error and stop executing but with more efficient error)

cc @adamaltman @tatomyr

**Comment by tatomyr** [comment]
I believe we should also cover `plugins` and `extends`. To me there's sense to proceed with a config which is trying to extend from non-existing config or a plugin. I believe we should add errors for those cases as well.

**Comment by adamaltman** [comment]
Yes, with a friendly error message.

File: packages/cli/src/utils.ts [hunk 6]
```diff
@@ -327,6 +356,24 @@ export function exitWithError(message: string) {
...
+export async function loadConfigAndHandleErrors(
+  options: {
+    configPath?: string;
+    customExtends?: string[];
+    processRawConfig?: (rawConfig: RawConfig) => void | Promise<void>;
+    files?: string[];
+    region?: Region;
+  } = {}
+): Promise<Config> {
+  let config = new Config({ apis: {}, styleguide: {} });
```
Comment (tatomyr) [inline packages/cli/src/utils.ts:372]: I'd suggest using `const` instead of `let`.

File: packages/core/src/config/config-resolvers.ts [hunk 7]
```diff
@@ -342,7 +346,7 @@ async function loadExtendStyleguideConfig(
 
     return rawConfig.styleguide;
   } catch (error) {
-    throw new Error(`Failed to load "${filePath}": ${error.message}`);
+    throw new Error(`Failed to load "${filePath}": Please provide a valid path`);
```
Comment (tatomyr) [inline packages/core/src/config/config-resolvers.ts:349]: Why don't we want to keep the generic `error.message` here? In future there might be different errors inside the function, not necessarily related to  a wrong path.

**Review by tatomyr — APPROVED** [review]

**Pull request merged**